ideascube/serveradmin/management/commands/catalog.py

```diff
@@ -0,0 +1,148 @@
+        list = subs.add_parser('list', help='List packages')
+        group = list.add_mutually_exclusive_group()
+        group.add_argument(
+            '--installed', action='store_true',
+            help='Only list installed packages')
+        group.add_argument(
+            '--available', action='store_true',
+            help='Only list available packages')
+        group.add_argument(
+            '--all', action='store_true',
+            help='List available and installed packages (default)')
+        list.set_defaults(func=self.list_packages)
+
+        install = subs.add_parser(
+            'install', parents=[required_ids], help='Install packages')
+        install.set_defaults(func=self.install_packages)
+
+        remove = subs.add_parser(
+            'remove', parents=[required_ids], help='Remove packages')
+        remove.set_defaults(func=self.remove_packages)
+
+        upgrade = subs.add_parser(
+            'upgrade', aliases=['update'], parents=[required_ids],
+            help='Upgrade packages')
+        upgrade.set_defaults(func=self.upgrade_packages)
+
+        # -- Manage local cache -----------------------------------------------
+        cache = subs.add_parser('cache', help='Manage cache')
+
+        cachesubs = cache.add_subparsers(title='Commands', dest='cachecmd')
+        cachesubs.required = True
+
+        update = cachesubs.add_parser('update', help='Update the local cache')
+        update.set_defaults(func=self.update_cache)
+
+        clear = cachesubs.add_parser('clear', help='Clear the local cache')
+        clear.set_defaults(func=self.clear_cache)
+
+        # -- Manage remote sources --------------------------------------------
+        remote = subs.add_parser('remotes', help='Manage remote sources')
+
+        remotesubs = remote.add_subparsers(title='Commands', dest='remotecmd')
+        remotesubs.required = True
+
+        list = remotesubs.add_parser('list', help='List known remote sources')
+        list.set_defaults(func=self.list_remotes)
+
+        add = remotesubs.add_parser('add', help='Add a remote source')
+        add.add_argument('id', help='The id for the new remote source')
+        add.add_argument('name', help='The name for the new remote source')
+        add.add_argument('url', help='The url for the new remote source')
+        add.set_defaults(func=self.add_remote)
+
+        rm = remotesubs.add_parser('remove', help='Remove a remote source')
+        rm.add_argument('id', help='The id of the remote source to remove')
+        rm.set_defaults(func=self.remove_remote)
+
+    def handle(self, *args, **options):
+        self.catalog = Catalog(settings.CATALOG_CACHE_BASE_DIR)
+        options['func'](options)
+
+    # -- Manage content -------------------------------------------------------
+    def list_packages(self, options):
+        if not any([options[k] for k in ('installed', 'available')]):
```

Comment:
Oh, so you meant two groups in the output!

Here I pretty much did what yum/dnf do, since I'm used to their output for quite a few years. :smile: 

One argument in favor of separating the two, is that we can show that `wikipedia.fr` is installed in its version `2015-08`, and it is also available but in a different, newer version. Oh hey, we need to update it!